src/common/config.cpp

```diff
@@ -11,10 +12,76 @@ namespace marian {
 
 size_t Config::seed = (size_t)time(0);
 
+Config::Config(const std::string options,
+               cli::mode mode /*= cli::mode::training*/,
+               bool validate /*= false*/) {
+  std::vector<std::string> sargv;
+  utils::Split(options, sargv, " ");
```

Comment:
What is the options string? This looks very dangerously as not parsing a command line correctly, w.r.t. escaping etc. Maybe this funcition needs a comment stating what it is for and in which context it is meant to be used.